src/transport/SessionManager.cpp

```diff
@@ -527,6 +527,7 @@ void SessionManager::UnauthenticatedMessageDispatch(const PacketHeader & packetH
     if (mCB != nullptr)
     {
         CHIP_TRACE_MESSAGE_RECEIVED(payloadHeader, packetHeader, unsecuredSession, peerAddress, msg->Start(), msg->TotalLength());
+        SessionHolder holder(session);
```

Comment:
Can you explain why this is needed? I thought that SessionHandle increases the reference count of a session (at least `UnauthenticatedSession`s have `Retain()/Release()` implemented, so a session can't be removed as long as a handle exists. What am I missing?

Reply:
I'm going to make `SessionHandle` not reference counted, because it is not allowed to store SessionHandle anywhere except for function arguments and return values. Your commend remind me that I should reconsider it, it may not a good idea to do so.

So let's keep `SessionHandle` reference counted, and close this PR.